execution_engine/src/core/runtime/mod.rs

```diff
@@ -2994,3 +3000,22 @@ where
         Ok(Ok(()))
     }
 }
+
+#[cfg(feature = "test-support")]
+fn dump_debug_info(error: &wasmi::Error, instance: wasmi::ModuleRef) {
+    // Currently, we're only interested in the details for the `Unreachable` error.
+
+    if let wasmi::Error::Trap(trap) = error {
+        if let TrapKind::Unreachable = trap.kind() {
+            eprintln!("Wasm execution error: TrapCode::Unreachable");
+            eprintln!(
+                "current instance stack height = {}",
+                if let Some(stack_height) = instance.globals().last() {
+                    format!("{:?}", stack_height.get())
+                } else {
+                    "unknown".to_string()
+                }
+            );
```

Comment:
As discussed offline, I think it would be less confusing if this function only printed an error referencing the stack height if the stack height exceeded the configured `max_stack_height`.  In that case, I'd prefer if the logged message was more like `stack height limit of {max_stack_height} exceeded`.  Not sure if there's even any point to printing the actual stack height, as I'd expect it should be `max_stack_height + 1`, but maybe worth confirming that?

Reply:
Updated in 7f8dcebbde140e9be8b10500c7fb9ac5ff355fe7